File: src/ast/ConditionalExpression.java
package ast;

import java.io.StringWriter;

public class ConditionalExpression extends BaseNode {
    private BaseNode _thenNode;
    private BaseNode _elseNode;
    private BaseNode _conditionNode;

    public ConditionalExpression(BaseNode conditionNode, BaseNode thenNode, BaseNode elseNode) {
        super(NodeType.ConditionalExpression);
        _thenNode = thenNode;
        _conditionNode = conditionNode;
        _elseNode = elseNode;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        writer.write("(");
        _conditionNode.Print(writer);
        writer.write(") ? (");
        _thenNode.Print(writer);
        writer.write(") : (");
        _elseNode.Print(writer);
        writer.write(")");
    }
}

File: src/de/fearlesstobi/demangler/ast/ConditionalExpression.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class ConditionalExpression(private val conditionNode: BaseNode, private val thenNode: BaseNode, private val elseNode: BaseNode) : BaseNode(NodeType.ConditionalExpression) {
    public override fun printLeft(writer: StringWriter) {
        writer.write("(")
        conditionNode.print(writer)
        writer.write(") ? (")
        thenNode.print(writer)
        writer.write(") : (")
        elseNode.print(writer)
        writer.write(")")
    }

}